agreement-component.js

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+define([], function() {
+
+'use strict';
+
+function register(module) {
+  module.component('brAgreement', {
+    controller: Ctrl,
+    bindings: {
+      onAccept: '&brOnAccept',
+      onCancel: '&brOnCancel'
+    },
+    templateUrl:
+      requirejs.toUrl('bedrock-angular-agreement/agreement-component.html')
+  });
+}
+
+/* @ngInject */
+function Ctrl($scope, brAgreementService) {
+  var self = this;
+  self.agreementHeader = brAgreementService.agreementHeader;
+  self.agreements = brAgreementService.agreements;
+  self.displayOrder = brAgreementService.displayOrder;
+  self.selectedAgreement = null;
+
+  $scope.$watch(function() {
```

Comment:
This should be implemented as an `ng-change` callback function `onAgree` in the template rather than a `$watch`. It should be more performant. Then, whenever it's called, check `allAgreed`, etc.